lineax/_operator.py

```diff
@@ -1363,10 +1363,40 @@ def diagonal(operator: AbstractLinearOperator) -> Shaped[Array, " size"]:
 
 @diagonal.register(MatrixLinearOperator)
 @diagonal.register(PyTreeLinearOperator)
```

Comment:
Side note, I think we could make the `PyTreeLinearOperator` case a little more efficient by calling `jnp.diag` on each 'diagonal leaf', and concatenate those together. (You don't have to do that here, just spotting it.)